### PR TITLE
History API.

### PR DESCRIPTION
**Pull request by hussiantabha**
Now when user watches a video that video will get stored in the history page.

### CODE REVIEW

File: src/pages/History.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    console.log(convertedJSON);
+    dispatch({
+      type: "historyVideos",
+      payload: { value: convertedJSON.history },
+    });
+  };
+  const deleteHistoryVideo = async (id) => {
+    try {
+      const deleteData = await fetch(`/api/user/history/${id}`, {
+        method: "DELETE",
```
Comment (shrushti2000) [inline src/pages/History.js:27]: These history service functions can be moved in a seperate file. Just a suggestion.

File: src/pages/History.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+          {videoState.history.map(
+            ({
+              _id,
+              title,
+              thumbnail,
+              avatar_url,
+              creator,
+              views,
+              uploaded,
+              duration,
```
Comment (shrushti2000) [inline src/pages/History.js:86]: Good that you have destructured video attributes here.

File: src/pages/History.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            }) => {
+              return (
+                <div className="video-card-container" key={_id}>
+                  <Link to={`/video/${_id}`}>
+                    <div className="video-img-container">
+                      <img src={thumbnail} className="video-img" />
```
Comment (shrushti2000) [inline src/pages/History.js:92]: alt attribute is missing.

File: src/css/displayvideo.css [hunk 4]
```diff
@@ -1,11 +1,18 @@
+.main-sidebar-flex {
+  display: flex;
+  justify-content: space-around;
+  margin-top: 1rem;
+}
 .video-container {
   width: 780px;
   height: 400px;
```
Comment (shrushti2000) [inline src/css/displayvideo.css:8]: can use relative units here

File: src/css/displayvideo.css [hunk 5]
```diff
@@ -1,11 +1,18 @@
...
-  border: 1px solid black;
+  /* border: 1px solid black; */
+  border-bottom: 1px solid #d1d5db;
   display: flex;
   flex-flow: column;
```
Comment (shrushti2000) [inline src/css/displayvideo.css:12]: can create a css utility class for flex,since it is used at most of the places. Just a suggestion

**Review by ajit-sharmaGH — APPROVED** [review]
Code is clean, great use of proper naming, you can merge it

**Review by shrushti2000 — APPROVED** [review]
The history functionality is working properly. Left few suggestions . If possible please add a screen recording in the PR. Overall everything looks good. Good work . Can be merged.

File: src/css/displayvideo.css [hunk 6]
```diff
@@ -115,3 +121,21 @@
     height: 400px;
   }
 } ; */
+.sidebar-video-card {
+  display: flex;
+  justify-content: flex-start;
+}
+.sidebar-img {
+  width: 200px;
```
Comment (Kothalkarpriya) [inline src/css/displayvideo.css:129]: For more than 10px you can use rem convention

**Review by Kothalkarpriya — APPROVED** [review]
Code is clean and understandable , GOod to merge

**Pull request merged**